crates/swc_css_parser/src/parser/at_rule.rs

```diff
@@ -182,71 +182,87 @@ where
-            tokens.push(self.input.bump()?.unwrap());
+            match cur!(self) {
+                // <semicolon-token>
+                // Return the at-rule.
+                tok!(";") => {
+                    self.input.bump()?;
 
-            let mut brace_cnt = 1;
-            loop {
-                if is!(self, "}") {
-                    brace_cnt -= 1;
-                    if brace_cnt == 0 {
-                        tokens.push(self.input.bump()?.unwrap());
-                        break;
-                    }
-                }
-                if is!(self, "{") {
-                    brace_cnt += 1;
-                }
+                    at_rule.prelude.span = span!(self, prelude_span.lo);
+                    at_rule.span = span!(self, at_rule_span.lo);
 
-                let token = self.input.bump()?;
-                match token {
-                    Some(token) => tokens.push(token),
-                    None => break,
-                }
-            }
-        } else {
-            loop {
-                if eat!(self, ";") {
-                    break;
+                    return Ok(AtRule::Unknown(at_rule));
                 }
+                // <{-token>
+                // Consume a simple block and assign it to the at-rule’s block. Return the at-rule.
+                tok!("{") => {
+                    at_rule.prelude.span = span!(self, prelude_span.lo);
 
-                if is!(self, EOF) {
-                    break;
-                }
-                let token = self.input.bump()?;
-                match token {
-                    Some(token) => tokens.push(token),
-                    None => break,
+                    self.input.bump()?;
+
+                    at_rule.block = Some(self.parse_simple_block('}')?);
+                    at_rule.span = span!(self, at_rule_span.lo);
+
+                    return Ok(AtRule::Unknown(at_rule));
                 }
-            }
+                // anything else
+                // Reconsume the current input token. Consume a component value. Append the returned
+                // value to the at-rule’s prelude.
+                _ => {
+                    let token = match self.parse_component_value()? {
+                        ComponentValue::SimpleBlock(i) => {
+                            at_rule.prelude.tokens.extend(i.value.tokens);
+                        }
+                        ComponentValue::Function(i) => {
+                            for val in i.value.into_iter() {
+                                match val {
+                                    Value::Lazy(token) => {
+                                        at_rule.prelude.tokens.extend(token.tokens);
+                                    }
+                                    _ => {
+                                        unreachable!();
+                                    }
+                                }
+                            }
+                        }
```

Comment:
It is dirty workaround, because we need more refactor, I show it in `1.`, it will be simplified and removed in future

Reply:
Resolved in this PR